File: app/src/main/java/com/egecius/rxjava2_demo_2/rx/subjects/PublishSubjectExamples.java
package com.egecius.rxjava2_demo_2.rx.subjects;

import java.util.List;

import io.reactivex.subjects.PublishSubject;

public class PublishSubjectExamples {

    PublishSubject<Integer> emitsEventsAndOnComplete(List<Integer> list) {

        PublishSubject<Integer> subject = PublishSubject.create();

        for (final Integer integer : list) {
            subject.onNext(integer);
        }

        subject.onComplete();

        return subject;
    }

}

File: app/src/main/java/com/egecius/rxjava2_demo_2/rx/subjects/PublishSubjectExamples.kt
package com.egecius.rxjava2_demo_2.rx.subjects

import io.reactivex.subjects.PublishSubject

class PublishSubjectExamples {

    fun emitsEventsAndOnComplete(list: List<Int>): PublishSubject<Int> {

        val subject = PublishSubject.create<Int>()

        for (integer in list) {
            subject.onNext(integer)
        }

        subject.onComplete()

        return subject
    }

}
